### PR TITLE
[WIP] SDIRK 4th order 4 stage method

### PR DESCRIPTION
**Pull request by saurabhkgp21**
@ChrisRackauckas 
- What should be the initial guess(`nlsolver.z`) for implicit method for each stage? 
- Since this method isn't completely singly diagonal implicit, I have used `update_W!` after changing `gamma` in third stage. Is it correct?

Threading will be added once this method is tested without threads.

### CODE REVIEW

File: src/OrdinaryDiffEq.jl [hunk 1]
```diff
@@ -177,7 +177,7 @@ module OrdinaryDiffEq
 
   export ImplicitEuler, ImplicitMidpoint, Trapezoid, TRBDF2, SDIRK2, Kvaerno3,
          KenCarp3, Cash4, Hairer4, Hairer42, SSPSDIRK2, Kvaerno4, Kvaerno5,
-         KenCarp4, KenCarp5
+         KenCarp4, KenCarp5, RK4Threaded
```
Comment (ChrisRackauckas) [inline src/OrdinaryDiffEq.jl]: this is a bad name since it's not RK4 threaded...

**Comment by ChrisRackauckas** [comment]
>What should be the initial guess(nlsolver.z) for implicit method for each stage?

For now do the zero guess. This is something that needs to be derived.

>Since this method isn't completely singly diagonal implicit, I have used update_W! after changing gamma in third stage. Is it correct?

yes

File: src/OrdinaryDiffEq.jl (hunk 1 above)
Comment (saurabhkgp21) [reply src/OrdinaryDiffEq.jl]: Will `DIRK44_2`(Diagonal Implicit RK 4th order, 4 stage, 2 parallel) be okay?

Comment (ChrisRackauckas) [reply src/OrdinaryDiffEq.jl]: what's the name of the author the tableau comes from?

Comment (saurabhkgp21) [reply src/OrdinaryDiffEq.jl]: Iserles and Nørsett, 1990

Comment (ChrisRackauckas) [reply src/OrdinaryDiffEq.jl]: Yeah, so I'd call it like IserNor4

**Pull request closed without merging**